source/Sagan-RDBMS/RDBMSRepository.class.st

```diff
@@ -167,8 +167,11 @@ RDBMSRepository >> transact: aBlock [
 RDBMSRepository >> update: aMutableDomainObject executing: aMonadycBlock [
 
 	self
-		transact:
-			[ :session | session modify: aMutableDomainObject in: [ aMonadycBlock value: aMutableDomainObject ] ].
+		transact: [ :session | 
+			session
+				modify: aMutableDomainObject
+				in: [ aMonadycBlock value: ( session refresh: aMutableDomainObject ) ]
```

Comment:
Looks neat that #refresh:, but don't know what it does actually. 
We used to query inside that transact for the object again using `SimpleQuery read: aMutableDomainObject 
 class where: [:object | object = aMutableDomainObject ]` instead of that #refresh:. 
Not saying is better; in fact, regarding performance probably the first implementation is worse. Only trying to understand the difference.

Reply:
Well, it seems not to be covered in Glorp official doc. However it is mentioned in VAST Glorp tutorial (https://www.instantiations.com/docs/92/db/wwhelp/wwhimpl/js/html/wwhelp.htm#href=dbglp03.html) section "Reading, Writing, Deleting Data" which says:
```
Rereading
When we read data from a database the data exists in two places: in the database and in memory. If another thread or program modifies the database, the data in memory will be different than what is in the database. The method GlorpSession>>refresh: will update an object with the current data in the database.

foundPerson := session readOneOf: Person where: [:each | each  firstName  = 'Sam'].
“some time passes”
session refresh: foundPerson
```
I think there is no difference, because that method internally executes a query accessing the table via PK. And sets the flag shouldRefresh to true before executing it.